pkg/k8sapi/discovery.go

```diff
@@ -68,15 +68,31 @@ type Controller struct {
 	controller *controller
 	kubeClient kubernetes.Interface
 	myNodeName string
-	synced     bool
 }
 
 // NewController creates a new DiscoveryController
-func NewController(clientset kubernetes.Interface) *Controller {
-	return &Controller{kubeClient: clientset,
+func NewController(clientset kubernetes.Interface) (*Controller, error) {
+	stopCh := wait.NeverStop
+
+	d := &Controller{kubeClient: clientset,
 		myNodeName: os.Getenv("MY_NODE_NAME"),
 		cniPods:    make(map[string]string),
 		workerPods: make(map[string]*K8SPodInfo)}
+
+	log.Info("Starting Pod informer")
+
+	go d.controller.informer.Run(stopCh)
+
+	log.Info("Waiting for controller cache sync")
+	// Wait for all involved caches to be synced, before processing items from the queue is started
+	if !cache.WaitForCacheSync(stopCh, d.controller.informer.HasSynced) {
```

Comment:
On a future PR (not this one), we should use the `WaitForNamedCacheSync` method that @mogren alluded to in an earlier comment:

https://github.com/kubernetes/client-go/blob/bc9b51d240b2f5000adc2fa96231ab219ab0d702/tools/cache/shared_informer.go#L196

Which would basically remove the need for the log lines here.